moveit_core/robot_state/test/robot_state_benchmark.cpp

```diff
@@ -66,312 +61,279 @@ Eigen::Isometry3d createTestIsometry()
+    benchmark::ClobberMemory();
   }
 }
 
 // Benchmark time to invert an Eigen::Matrix4d.
 static void inverseMatrix4d(benchmark::State& st)
 {
-  int n_iters = st.range(0);
-  Eigen::Isometry3d isometry = createTestIsometry();
-  Eigen::Affine3d affine;
-  affine.matrix() = isometry.matrix();
-
+  Eigen::Affine3d affine = createTestIsometry();
+  Eigen::Affine3d result;
   for (auto _ : st)
   {
-    for (int i = 0; i < n_iters; ++i)
-    {
-      Eigen::Affine3d result;
-      benchmark::DoNotOptimize(result = affine.matrix().inverse());
-      benchmark::ClobberMemory();
-    }
+    benchmark::DoNotOptimize(result = affine.matrix().inverse());
+    benchmark::ClobberMemory();
   }
 }
 
-// Benchmark time to construct a RobotState given a RobotModel.
-static void robotStateConstruct(benchmark::State& st)
+struct RobotStateBenchmark : ::benchmark::Fixture
 {
-  int n_states = st.range(0);
-  const moveit::core::RobotModelPtr& robot_model = moveit::core::loadTestingRobotModel(PANDA_TEST_ROBOT);
-
-  // Make sure the group exists, otherwise exit early with an error.
-  if (!robot_model->hasJointModelGroup(PANDA_TEST_GROUP))
+  void SetUp(const ::benchmark::State& /*state*/) override
   {
-    st.SkipWithError("The planning group doesn't exist.");
-    return;
-  }
-
-  for (auto _ : st)
-  {
-    for (int i = 0; i < n_states; i++)
+    if (rcutils_logging_set_logger_level("moveit_robot_model.robot_model", RCUTILS_LOG_SEVERITY_WARN) != RCUTILS_RET_OK)
     {
-      std::unique_ptr<moveit::core::RobotState> robot_state;
-      benchmark::DoNotOptimize(robot_state = std::make_unique<moveit::core::RobotState>(robot_model));
-      benchmark::ClobberMemory();
     }
+
+    robot_model = moveit::core::loadTestingRobotModel(PANDA_TEST_ROBOT);
   }
-}
 
-// Benchmark time to copy a RobotState.
-static void robotStateCopy(benchmark::State& st)
-{
-  int n_states = st.range(0);
-  const moveit::core::RobotModelPtr& robot_model = moveit::core::loadTestingRobotModel(PANDA_TEST_ROBOT);
+  std::vector<moveit::core::RobotState> constructStates(size_t num)
+  {
+    std::vector<moveit::core::RobotState> states;
+    states.reserve(num);
+    for (size_t i = 0; i < num; i++)
+      states.emplace_back(robot_model);
+    return states;
+  }
+  std::vector<moveit::core::RobotState> constructStates(size_t num, const moveit::core::RobotState& state)
+  {
+    std::vector<moveit::core::RobotState> states;
+    states.reserve(num);
+    for (size_t i = 0; i < num; i++)
+      states.emplace_back(state);
+    return states;
+  }
 
-  // Make sure the group exists, otherwise exit early with an error.
-  if (!robot_model->hasJointModelGroup(PANDA_TEST_GROUP))
+  std::vector<size_t> randomPermudation(size_t num)
   {
-    st.SkipWithError("The planning group doesn't exist.");
-    return;
+    std::vector<size_t> result;
+    result.reserve(num);
+    for (size_t i = 0; i < num; i++)
+      result.push_back(i);
+
+    std::random_device random_device;
```

Comment:
What do you think about using a deterministic random sequence instead?
Not sure if it will change anything, but at least it would create more deterministic conditions.